phoenix-load-balancer/src/main/java/org/apache/phoenix/loadbalancer/service/LoadBalancer.java

```diff
@@ -0,0 +1,73 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.loadbalancer.service;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.phoenix.loadbalancer.zookeeper.ZookeeperServiceDiscoverer;
+import org.apache.phoenix.query.QueryServices;
+
+public class LoadBalancer {
+
+    protected static final Log LOG = LogFactory.getLog(LoadBalancer.class);
+    private static Configuration configuration;
+    private static ZookeeperServiceDiscoverer zookeeperServiceDiscoverer ;
+    private static LoadBalancer loadBalancer;
+    private static final String connectString;
+    private static String basePath ;
+    private static String serviceName;
+
+    static {
```

Comment:
How about moving this logic into a method that can lazily initialize `loadBalancer` instead of doing it in a static block?